raft/log.go

```diff
@@ -211,20 +216,26 @@ func (l *raftLog) term(i uint64) uint64 {
 		return t
 	}
 	if err == ErrCompacted {
-		return 0
+		return math.MaxUint64
```

Comment:
MaxUint64 seems like a risky return value. If it's ever used, it will be greater than any existing term, and will then immediately wrap around. What is the motivation for this part of the change?